clients/mongoStoreClient_test.go

```diff
@@ -290,6 +298,51 @@ func TestGetTimeLastEntryUserAndDevice_NoData(t *testing.T) {
 	if err != nil {
 		t.Fatalf("GetTimeLastEntryUserAndDevice unexpected error [%s]", err.Error())
 	}
+}
+
+func TestSpecifiedSchemaVersion(t *testing.T) {
+
+	allBasals := &model.QueryData{
+		MetaQuery: map[string]string{"userid": valid_userid},
+		Types:     []string{"basal"},
+		Sort:      map[string]string{"time": "myTime"},
+	}
+
+	type found map[string]interface{}
+
+	//default is all data greater or equal to version 0 matters
+	mc0 := initTestData(t, initConfig(all_schema_versions))
+
+	resultsV0, _ := mc0.ExecuteQuery(allBasals)
+
+	basalsV0 := []found{}
+	json.Unmarshal(resultsV0, &basalsV0)
+
+	if len(basalsV0) != 12 {
+		t.Fatalf("We should have 12 entries for _schemaVersion >= [%d] but got [%d]", mc0.config.SchemaVersionGte, len(basalsV0))
+	}
+
+	//default is all data greater or equal to version 99 matters
+	mc99 := initTestData(t, initConfig(99))
+	resultsV99, _ := mc99.ExecuteQuery(allBasals)
+
+	basalsV99 := []found{}
+	json.Unmarshal(resultsV99, &basalsV99)
+
+	if len(basalsV99) != 4 {
+		t.Fatalf("We should have 4 entries for _schemaVersion >= [%d] but got [%d]", mc99.config.SchemaVersionGte, len(basalsV99))
+	}
+
+	//now only data greater or equal to version 101 matters
+	mc101 := initTestData(t, initConfig(101))
+	resultsV101, _ := mc101.ExecuteQuery(allBasals)
+
+	basalsV101 := []found{}
+	json.Unmarshal(resultsV101, &basalsV101)
+
+	if len(basalsV101) != 0 {
+		t.Fatalf("We should have 4 entries for _schemaVersion >= [%d] but got [%d]", mc101.config.SchemaVersionGte, len(basalsV101))
```

Comment:
Random question having absolutely nothing to do with this PR:

What is the intent with putting some log values in square brackets?